presto-native-execution/presto_cpp/main/PrestoServer.cpp

```diff
@@ -906,22 +917,30 @@ void PrestoServer::stop() {
       << "Waiting for " << shutdownOnsetSec
       << " second(s) before proceeding with the shutdown...";
 
-  // Give coordinator some time to receive our new node state and stop sending
-  // any tasks.
-  std::this_thread::sleep_for(std::chrono::seconds(shutdownOnsetSec));
+  std::thread shutdownThread([this, shutdownOnsetSec]() {
```

Comment:
what is the motivation for a separate shutdown thread?

Reply:
We were not able to read the server status if we didn't schedule the shutdown in a separate thread because before we could read the status, server was shutting down.

Reply:
We need it to compare it against the status codes in the test cases.

Reply:
Again not ideal to make changes for testing.